src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -148,13 +148,13 @@ public StreamReader(Stream stream, Encoding? encoding = null, bool detectEncodin
             {
                 throw new ArgumentException(SR.Argument_StreamNotReadable);
             }
-            if (bufferSize == -1)
+            if (bufferSize == -1 || bufferSize == 0)
```

Comment:
With FileStream, bufferSize == 0 means "don't buffer".  Here it's being made to mean "use the default buffer size"?

Reply:
Yes, here it's "use the default buffer size". Moreover, if the user provides a buffer size smaller than `MinBufferSize` it's used instead.

https://github.com/dotnet/runtime/blob/fca27371965445e149144cc50775f5ee63367e08/src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs#L163-L166

To be honest with you I am not 100% convinced that we should change it. What is your opinion on that?

Reply:
I think we should leave it the way it was, with 0 throwing.

Before everything around 0 was consistent: 0 as the buffer size for file-related stuff threw an exception.  A recent behavioral change to FileStream was made to allow 0 to mean "don't buffer", creating an inconsistency.  If the change here in StreamReader were to apply the same "don't buffer" semantics as in FileStream, then I think it'd be reasonable, but instead it's extending the inconsistency to now mean something different in StreamReader than it does in FileStream, and with no exception to indicate the difference.  I don't think the change is making things better.

My $.02.

Reply:
Thank you for the feedback. I've reverted the change.